misk/src/main/kotlin/misk/web/marshal/Generic.kt

```diff
@@ -0,0 +1,109 @@
+package misk.web.marshal
+
+import com.google.inject.TypeLiteral
+import misk.inject.typeLiteral
+import misk.web.Response
+import misk.web.ResponseBody
+import okhttp3.MediaType
+import okio.BufferedSink
+import okio.BufferedSource
+import okio.ByteString
+import java.lang.reflect.ParameterizedType
+import kotlin.reflect.KParameter
+
+/** Handles generic unmarshalling, for cases where the action can accept anything */
+object GenericUnmarshallers {
+    val GENERIC_REQUEST_TYPES = setOf(
+            String::class.java,
+            Array<Byte>::class.java,
+            BufferedSource::class.java,
+            ByteString::class.java
+    )
+
+    object ToString : Unmarshaller {
+        override fun unmarshal(source: BufferedSource): Any? = source.readUtf8()
+    }
+
+    object ToByteArray : Unmarshaller {
+        override fun unmarshal(source: BufferedSource): Any? = source.readByteArray()
+    }
+
+    object ToBufferedSource : Unmarshaller {
+        override fun unmarshal(source: BufferedSource): Any? = source
+    }
+
+    object ToByteString : Unmarshaller {
```

Comment:
Doesn't seem like these are used outside this class, can they be private?

Reply:
yep